TravelAgency-service/src/main/java/cz/muni/fi/travelAgency/facade/ExcursionFacadeImpl.java

```diff
@@ -25,17 +28,23 @@
     @Autowired
     private ExcursionService excursionService;
 
+    @Autowired
+    private TripService tripService;
+
     /**
      * Mapper responsible for mapping DTOs to Entities.
      */
     @Autowired
     private BeanMappingService beanMappingService;
 
     @Override
-    public void createExcursion(ExcursionDTO excursionDTO) {
+    public void createExcursion(ExcursionCreateDTO excursionDTO) {
         Excursion mappedExcursion = beanMappingService.mapTo(excursionDTO, Excursion.class);
-        excursionService.createExcursion(mappedExcursion);
-        excursionDTO.setId(mappedExcursion.getId());
+        mappedExcursion.setExcursionDuration(Duration.ofMinutes(excursionDTO.getDurationMinutes()));
+        mappedExcursion.setTrip(tripService.findById(excursionDTO.getTripId()));
+
+        Excursion excursion = excursionService.createExcursion(mappedExcursion);
+        excursionDTO.setId(excursion.getId());
```

Comment:
Here just use ID from the mappedExcursion please. It will be updated by the create method. No need to return.